metrics/files/grafana/overview.js

```diff
@@ -258,6 +312,30 @@ function row_of_node_panels(node,prefix) {
       panel_collectd_ntp("Time",prefix,node),
       panel_collectd_logstash_event_types("Events",node)
     ]
+  } else {
+    omit_columns = arg_omit_columns.split(',')
+    if ( omit_columns.indexOf("CPU") === -1 ) {
+      valid_panels.push( panel_collectd_delta_cpu("CPU",prefix,node) )
+    }
+    if ( omit_columns.indexOf("Load") === -1 ) {
+      valid_panels.push( panel_collectd_loadavg("Load",prefix,node) )
+    }
+    if ( omit_columns.indexOf("Memory") === -1 ) {
+      valid_panels.push( panel_collectd_memory("Memory",prefix,node) )
+    }
+    if ( omit_columns.indexOf("Time") === -1 ) {
+      valid_panels.push( panel_collectd_ntp("Time",prefix,node) )
+    }
+    if ( omit_columns.indexOf("Events") === -1 ) {
+      valid_panels.push( panel_collectd_logstash_event_types("Events",node) )
+    }
+  }
```

Comment:
I thought about that, but the function names and definitions vary, so it
wasn't easy.

TBH this works for now, ripe for refactorink later.


On 4 February 2015 at 16:50, Matthew Mead-Briggs <notifications@github.com>
wrote:

> In metrics/files/grafana/overview.js
> <https://github.com/ministryofjustice/metrics-formula/pull/45#discussion_r24098013>
> :
>
> > +    if ( omit_columns.indexOf("CPU") === -1 ) {
> > +      valid_panels.push( panel_collectd_delta_cpu("CPU",prefix,node) )
> > +    }
> > +    if ( omit_columns.indexOf("Load") === -1 ) {
> > +      valid_panels.push( panel_collectd_loadavg("Load",prefix,node) )
> > +    }
> > +    if ( omit_columns.indexOf("Memory") === -1 ) {
> > +      valid_panels.push( panel_collectd_memory("Memory",prefix,node) )
> > +    }
> > +    if ( omit_columns.indexOf("Time") === -1 ) {
> > +      valid_panels.push( panel_collectd_ntp("Time",prefix,node) )
> > +    }
> > +    if ( omit_columns.indexOf("Events") === -1 ) {
> > +      valid_panels.push( panel_collectd_logstash_event_types("Events",node) )
> > +    }
> > +  }
>
> Would it be easy to do this in a loop? pseudo:
>
> for panel in valid_panels:
>     if valid_panels not in omit_columns:
>          ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ministryofjustice/metrics-formula/pull/45/files#r24098013>
> .
>